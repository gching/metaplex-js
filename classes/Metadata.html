<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>Metadata | metaplex-js</title>
	<meta name="description" content="Documentation for metaplex-js">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="../assets/css/main.css">
	<script async src="../assets/js/search.js" id="search-script"></script>
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="../assets/js/search.json" data-base="..">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="../index.html" class="title">metaplex-js</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="../modules.html">metaplex-js</a>
				</li>
				<li>
					<a href="Metadata.html">Metadata</a>
				</li>
			</ul>
			<h1>Class Metadata</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel tsd-hierarchy">
				<h3>Hierarchy</h3>
				<ul class="tsd-hierarchy">
					<li>
						<span class="target">Metadata</span>
					</li>
				</ul>
			</section>
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Constructors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-constructor tsd-parent-kind-class"><a href="Metadata.html#constructor" class="tsd-kind-icon">constructor</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Properties</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Metadata.html#accountPublicKey" class="tsd-kind-icon">account<wbr>Public<wbr>Key</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Metadata.html#data" class="tsd-kind-icon">data</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Metadata.html#edition" class="tsd-kind-icon">edition</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Metadata.html#editionKey" class="tsd-kind-icon">edition<wbr>Key</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Metadata.html#isMutable" class="tsd-kind-icon">is<wbr>Mutable</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Metadata.html#key" class="tsd-kind-icon">key</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Metadata.html#mint" class="tsd-kind-icon">mint</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Metadata.html#primarySaleHappened" class="tsd-kind-icon">primary<wbr>Sale<wbr>Happened</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Metadata.html#updateAuthority" class="tsd-kind-icon">update<wbr>Authority</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><a href="Metadata.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Accessors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-get-signature tsd-parent-kind-class"><a href="Metadata.html#isMutableBool" class="tsd-kind-icon">is<wbr>Mutable<wbr>Bool</a></li>
								<li class="tsd-kind-get-signature tsd-parent-kind-class"><a href="Metadata.html#isPrimarySaleHappened" class="tsd-kind-icon">is<wbr>Primary<wbr>Sale<wbr>Happened</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Methods</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-method tsd-parent-kind-class"><a href="Metadata.html#getEditionKey" class="tsd-kind-icon">get<wbr>Edition<wbr>Key</a></li>
								<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><a href="Metadata.html#decode" class="tsd-kind-icon">decode</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Constructors</h2>
				<section class="tsd-panel tsd-member tsd-kind-constructor tsd-parent-kind-class">
					<a name="constructor" class="tsd-anchor"></a>
					<h3>constructor</h3>
					<ul class="tsd-signatures tsd-kind-constructor tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon">new <wbr>Metadata<span class="tsd-signature-symbol">(</span>args<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Omit</span><span class="tsd-signature-symbol">&lt;</span><a href="Metadata.html" class="tsd-signature-type" data-tsd-kind="Class">Metadata</a><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">&quot;key&quot;</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">&quot;isPrimarySaleHappened&quot;</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">&quot;isMutableBool&quot;</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">&quot;getEditionKey&quot;</span><span class="tsd-signature-symbol">&gt;</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="Metadata.html" class="tsd-signature-type" data-tsd-kind="Class">Metadata</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L48">models/programs/token-metadata/Metadata.ts:48</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>args: <span class="tsd-signature-type">Omit</span><span class="tsd-signature-symbol">&lt;</span><a href="Metadata.html" class="tsd-signature-type" data-tsd-kind="Class">Metadata</a><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">&quot;key&quot;</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">&quot;isPrimarySaleHappened&quot;</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">&quot;isMutableBool&quot;</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">&quot;getEditionKey&quot;</span><span class="tsd-signature-symbol">&gt;</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="Metadata.html" class="tsd-signature-type" data-tsd-kind="Class">Metadata</a></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Properties</h2>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="accountPublicKey" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagOptional">Optional</span> account<wbr>Public<wbr>Key</h3>
					<div class="tsd-signature tsd-kind-icon">account<wbr>Public<wbr>Key<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L31">models/programs/token-metadata/Metadata.ts:31</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="data" class="tsd-anchor"></a>
					<h3>data</h3>
					<div class="tsd-signature tsd-kind-icon">data<span class="tsd-signature-symbol">:</span> <a href="Data.html" class="tsd-signature-type" data-tsd-kind="Class">Data</a></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L34">models/programs/token-metadata/Metadata.ts:34</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="edition" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagOptional">Optional</span> edition</h3>
					<div class="tsd-signature tsd-kind-icon">edition<span class="tsd-signature-symbol">:</span> <a href="Edition.html" class="tsd-signature-type" data-tsd-kind="Class">Edition</a><span class="tsd-signature-symbol"> | </span><a href="MasterEditionV1.html" class="tsd-signature-type" data-tsd-kind="Class">MasterEditionV1</a><span class="tsd-signature-symbol"> | </span><a href="MasterEditionV2.html" class="tsd-signature-type" data-tsd-kind="Class">MasterEditionV2</a></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L38">models/programs/token-metadata/Metadata.ts:38</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="editionKey" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagOptional">Optional</span> edition<wbr>Key</h3>
					<div class="tsd-signature tsd-kind-icon">edition<wbr>Key<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L37">models/programs/token-metadata/Metadata.ts:37</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="isMutable" class="tsd-anchor"></a>
					<h3>is<wbr>Mutable</h3>
					<div class="tsd-signature tsd-kind-icon">is<wbr>Mutable<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L36">models/programs/token-metadata/Metadata.ts:36</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="key" class="tsd-anchor"></a>
					<h3>key</h3>
					<div class="tsd-signature tsd-kind-icon">key<span class="tsd-signature-symbol">:</span> <a href="../enums/MetadataKey.html" class="tsd-signature-type" data-tsd-kind="Enumeration">MetadataKey</a><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L30">models/programs/token-metadata/Metadata.ts:30</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="mint" class="tsd-anchor"></a>
					<h3>mint</h3>
					<div class="tsd-signature tsd-kind-icon">mint<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L33">models/programs/token-metadata/Metadata.ts:33</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="primarySaleHappened" class="tsd-anchor"></a>
					<h3>primary<wbr>Sale<wbr>Happened</h3>
					<div class="tsd-signature tsd-kind-icon">primary<wbr>Sale<wbr>Happened<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L35">models/programs/token-metadata/Metadata.ts:35</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="updateAuthority" class="tsd-anchor"></a>
					<h3>update<wbr>Authority</h3>
					<div class="tsd-signature tsd-kind-icon">update<wbr>Authority<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L32">models/programs/token-metadata/Metadata.ts:32</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class tsd-is-static">
					<a name="SCHEMA" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> SCHEMA</h3>
					<div class="tsd-signature tsd-kind-icon">SCHEMA<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">BorshSchema</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L12">models/programs/token-metadata/Metadata.ts:12</a></li>
						</ul>
					</aside>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Accessors</h2>
				<section class="tsd-panel tsd-member tsd-kind-get-signature tsd-parent-kind-class">
					<a name="isMutableBool" class="tsd-anchor"></a>
					<h3>is<wbr>Mutable<wbr>Bool</h3>
					<ul class="tsd-signatures tsd-kind-get-signature tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon"><span class="tsd-signature-symbol">get</span> isMutableBool<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L44">models/programs/token-metadata/Metadata.ts:44</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">boolean</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-get-signature tsd-parent-kind-class">
					<a name="isPrimarySaleHappened" class="tsd-anchor"></a>
					<h3>is<wbr>Primary<wbr>Sale<wbr>Happened</h3>
					<ul class="tsd-signatures tsd-kind-get-signature tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon"><span class="tsd-signature-symbol">get</span> isPrimarySaleHappened<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L40">models/programs/token-metadata/Metadata.ts:40</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">boolean</span></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Methods</h2>
				<section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class">
					<a name="getEditionKey" class="tsd-anchor"></a>
					<h3>get<wbr>Edition<wbr>Key</h3>
					<ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon">get<wbr>Edition<wbr>Key<span class="tsd-signature-symbol">(</span>metadataProgramKey<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">PublicKey</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">PublicKey</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L57">models/programs/token-metadata/Metadata.ts:57</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>metadataProgramKey: <span class="tsd-signature-type">PublicKey</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">PublicKey</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class tsd-is-static">
					<a name="decode" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> decode</h3>
					<ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class tsd-is-static">
						<li class="tsd-signature tsd-kind-icon">decode<span class="tsd-signature-symbol">(</span>accountWithPublicKey<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">AccountWithPublicKey</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="Metadata.html" class="tsd-signature-type" data-tsd-kind="Class">Metadata</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-metadata/Metadata.ts#L79">models/programs/token-metadata/Metadata.ts:79</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>accountWithPublicKey: <span class="tsd-signature-type">AccountWithPublicKey</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="Metadata.html" class="tsd-signature-type" data-tsd-kind="Class">Metadata</a></h4>
						</li>
					</ul>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class=" ">
						<a href="../modules.html">Exports</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
				<ul class="current">
					<li class="current tsd-kind-class">
						<a href="Metadata.html" class="tsd-kind-icon">Metadata</a>
						<ul>
							<li class=" tsd-kind-constructor tsd-parent-kind-class">
								<a href="Metadata.html#constructor" class="tsd-kind-icon">constructor</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Metadata.html#accountPublicKey" class="tsd-kind-icon">account<wbr>Public<wbr>Key</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Metadata.html#data" class="tsd-kind-icon">data</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Metadata.html#edition" class="tsd-kind-icon">edition</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Metadata.html#editionKey" class="tsd-kind-icon">edition<wbr>Key</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Metadata.html#isMutable" class="tsd-kind-icon">is<wbr>Mutable</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Metadata.html#key" class="tsd-kind-icon">key</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Metadata.html#mint" class="tsd-kind-icon">mint</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Metadata.html#primarySaleHappened" class="tsd-kind-icon">primary<wbr>Sale<wbr>Happened</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Metadata.html#updateAuthority" class="tsd-kind-icon">update<wbr>Authority</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class tsd-is-static">
								<a href="Metadata.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a>
							</li>
							<li class=" tsd-kind-get-signature tsd-parent-kind-class">
								<a href="Metadata.html#isMutableBool" class="tsd-kind-icon">is<wbr>Mutable<wbr>Bool</a>
							</li>
							<li class=" tsd-kind-get-signature tsd-parent-kind-class">
								<a href="Metadata.html#isPrimarySaleHappened" class="tsd-kind-icon">is<wbr>Primary<wbr>Sale<wbr>Happened</a>
							</li>
							<li class=" tsd-kind-method tsd-parent-kind-class">
								<a href="Metadata.html#getEditionKey" class="tsd-kind-icon">get<wbr>Edition<wbr>Key</a>
							</li>
							<li class=" tsd-kind-method tsd-parent-kind-class tsd-is-static">
								<a href="Metadata.html#decode" class="tsd-kind-icon">decode</a>
							</li>
						</ul>
					</li>
				</ul>
				<ul class="after-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="../assets/js/main.js"></script>
</body>
</html>