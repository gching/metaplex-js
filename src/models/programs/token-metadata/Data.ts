import { BorshSchema } from '../../utils/types';
import { Creator } from './Creator';

export class Data {
  static SCHEMA: BorshSchema = new Map([
    [
      Data,
      {
        kind: 'struct',
        fields: [
          ['name', 'string'],
          ['symbol', 'string'],
          ['uri', 'string'],
          ['sellerFeeBasisPoints', 'u16'],
          ['creators', { kind: 'option', type: [Creator] }],
        ],
      },
    ],
    ...Creator.SCHEMA,
  ]);

  name: string;
  symbol: string;
  uri: string;
  sellerFeeBasisPoints: number;
  creators: Creator[] | undefined;
  constructor(args: {
    name: string;
    symbol: string;
    uri: string;
    sellerFeeBasisPoints: number;
    creators: Creator[] | undefined;
  }) {
    this.name = args.name;
    this.symbol = args.symbol;
    this.uri = args.uri;
    this.sellerFeeBasisPoints = args.sellerFeeBasisPoints;
    this.creators = args.creators;
  }
}
