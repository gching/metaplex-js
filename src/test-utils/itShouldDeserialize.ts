import { PublicKey } from '@solana/web3.js';
import { serialize } from 'borsh';
import { BorshSchema } from '../models/utils/types';

export function itShouldDeserialize<E>(
  cls: {
    SCHEMA: BorshSchema;
    // eslint-disable-next-line @typescript-eslint/ban-types
    decode: Function;
  },
  received: E,
  accountPubKey = PublicKey.default,
): void {
  const data = Buffer.from(serialize(cls.SCHEMA, received));

  const deserialized = cls.decode({
    pubkey: accountPubKey,
    account: {
      data,
    },
  }) as E;

  expect(deserialized).toDeepEqual(received);
}
