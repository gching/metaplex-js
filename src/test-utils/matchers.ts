import { isEqualWith } from 'lodash';
import BN from 'bn.js';

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace jest {
    interface Matchers<R> {
      toDeepEqual: <E = any>(expected: E) => R;
    }
  }
}

expect.extend({
  toDeepEqual<E = any>(received: E, expected: E) {
    const pass = isEqualWith(received, expected, (objValue, otherValue) => {
      if (BN.isBN(objValue) && BN.isBN(otherValue)) {
        return objValue.eq(otherValue);
      }
      return undefined;
    });

    if (pass) {
      return {
        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
        message: () => `expected ${received} not to deep equal ${expected}`,
        pass: true,
      };
    }

    return {
      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
      message: () => `expected ${received} to deep equal ${expected}`,
      pass: false,
    };
  },
});

export default undefined;
