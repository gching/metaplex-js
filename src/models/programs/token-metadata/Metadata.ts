import { PublicKey } from '@solana/web3.js';
import { deserializeUnchecked } from 'borsh';
import { BorshSchema } from '../../utils/types';
import { Data } from './Data';
import { Edition } from './Edition';
import { MasterEditionV1 } from './MasterEditionV1';
import { MasterEditionV2 } from './MasterEditionV2';
import { MetadataKey } from './MetadataKey';
import { AccountWithPublicKey } from '../../../utils/privateTypes';

export class Metadata {
  static SCHEMA: BorshSchema = new Map([
    [
      Metadata,
      {
        kind: 'struct',
        fields: [
          ['key', 'u8'],
          ['updateAuthority', 'pubkey'],
          ['mint', 'pubkey'],
          ['data', Data],
          ['primarySaleHappened', 'u8'], // bool
          ['isMutable', 'u8'], // bool
        ],
      },
    ],
    ...Data.SCHEMA,
  ]);

  key: MetadataKey = MetadataKey.MetadataV1;
  accountPublicKey?: PublicKey;
  updateAuthority: PublicKey;
  mint: PublicKey;
  data: Data;
  primarySaleHappened: 0 | 1;
  isMutable: 0 | 1;
  editionKey?: PublicKey;
  edition?: Edition | MasterEditionV1 | MasterEditionV2;

  get isPrimarySaleHappened(): boolean {
    return this.primarySaleHappened === 1;
  }

  get isMutableBool(): boolean {
    return this.isMutable === 1;
  }

  constructor(
    args: Omit<
      Metadata,
      'key' | 'isPrimarySaleHappened' | 'isMutableBool' | 'getEditionKey'
    >,
  ) {
    Object.assign(this, args);
  }

  async getEditionKey(metadataProgramKey: PublicKey): Promise<PublicKey> {
    if (this.editionKey) {
      return this.editionKey;
    }

    const seeds = [
      Buffer.from('metadata'),
      metadataProgramKey.toBuffer(),
      this.mint.toBuffer(),
      Buffer.from('edition'),
    ];

    const editionKey = await PublicKey.findProgramAddress(
      seeds,
      metadataProgramKey,
    );

    this.editionKey = editionKey[0];

    return this.editionKey;
  }

  static decode(accountWithPublicKey: AccountWithPublicKey): Metadata {
    const metadata = deserializeUnchecked(
      Metadata.SCHEMA,
      Metadata,
      accountWithPublicKey.account.data,
    ) as Metadata;

    return new Metadata({
      ...metadata,
      accountPublicKey: accountWithPublicKey.pubkey,
    });
  }
}
