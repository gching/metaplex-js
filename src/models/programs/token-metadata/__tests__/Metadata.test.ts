import { PublicKey } from '@solana/web3.js';
import { itShouldDeserialize } from '../../../../test-utils/itShouldDeserialize';
import { Metadata } from '../Metadata';
import { Data } from '../Data';

describe('Metadata', () => {
  it('should instantiate with the correct values', () => {
    const inst = new Metadata({
      accountPublicKey: PublicKey.default,
      updateAuthority: PublicKey.default,
      data: new Data({
        uri: '1',
        sellerFeeBasisPoints: 0,
        name: 'test',
        symbol: 'test',
        creators: [],
      }),
      mint: PublicKey.default,
      isMutable: 0,
      primarySaleHappened: 1,
    });

    expect(inst).toMatchSnapshot();
  });

  it('should deserialize with the correct values', () => {
    const inst = new Metadata({
      accountPublicKey: PublicKey.default,
      updateAuthority: PublicKey.default,
      data: new Data({
        uri: '1',
        sellerFeeBasisPoints: 0,
        name: 'test',
        symbol: 'test',
        creators: [],
      }),
      mint: PublicKey.default,
      isMutable: 0,
      primarySaleHappened: 1,
    });

    itShouldDeserialize(Metadata, inst);
  });
});
