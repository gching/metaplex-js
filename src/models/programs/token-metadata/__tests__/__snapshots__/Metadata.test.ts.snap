// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metadata should instantiate with the correct values 1`] = `
Metadata {
  "accountPublicKey": PublicKey {
    "_bn": "00",
  },
  "data": Data {
    "creators": Array [],
    "name": "test",
    "sellerFeeBasisPoints": 0,
    "symbol": "test",
    "uri": "1",
  },
  "isMutable": 0,
  "key": 4,
  "mint": PublicKey {
    "_bn": "00",
  },
  "primarySaleHappened": 1,
  "updateAuthority": PublicKey {
    "_bn": "00",
  },
}
`;
